packages/network/src/connection/webrtc/WebRtcConnection.ts

```diff
@@ -417,10 +417,13 @@ export abstract class WebRtcConnection extends ConnectionEmitter {
                 let sent = false
                 let isOpen
                 try {
+                    if (this.isOpen()) {
+                        this.doSendMessage(queueItem.getMessage())
+                        sent = true
+                    }
                     // this.isOpen() is checked immediately after the call to node-datachannel.sendMessage() as if
                     // this.isOpen() returns false after a "successful" send, the message is lost with a near 100% chance.
                     // This does not work as expected if this.isOpen() is checked before sending a message
-                    sent = this.isOpen() && this.doSendMessage(queueItem.getMessage())
                     isOpen = this.isOpen()
                     sent = sent && isOpen
                     this.messagesSent += 1
```

Comment:
If `sent` is false, we should not increase `messageSent` and `bytesSent`. Also we could move the increment of sendFailures into the else branch and remove the `return`.

Maybe in a separate PR.